service/common.go

```diff
@@ -14,3 +14,18 @@ type AwsCredentials struct {
 	AwsProfile         string `json:"aws_profile,omitempty"`
 	AwsRegion          string `json:"aws_region,omitempty"`
 }
+
+// UsernameAndPassword represents a username and password. Used for basic auth, for example, MySQL
+// username and password in a database upstream configuration.
+type UsernameAndPassword struct {
```

Comment:
I think this one makes sense to use different username, and password. Because some sockets will have only username for example (e.g. standard ss with private key), some others will have username-provider, and username... and so on.

Reply:
Yeah. When using IAM auth in database socket, only username is needed. This is just for code reuse. It's embedded in a couple of structs that need username and password fields.